### ISSUE TITLE
Does browserstack work with socket.io v3?

### ISSUE THREAD

**Issue by johnjbarton** [karma-runner/karma-browserstack-launcher#186]
After we updated karma to socket io v3, BrowserStack fails on IE and Safari:
https://github.com/karma-runner/karma/commit/1b9e1de7d081e1c205debff27c6b5e1fe0585dee

Works on Firefox and Chrome.

Any hints?

**Comment by wati123321** [comment]
Do you encounter an error in your local console for this?

**Comment by johnjbarton** [comment]
Sorry I do not understand "local console".  

The tests pass on my MacBook Air running against chrome and firefox. They pass on Travis running against browserstack chrome and firefox. They newly fail on Travis against browserstack IE and safari.  I assume the difference is eg virtual machines or such.

**Comment by johnjbarton** [comment]
Any news?  We're stuck

**Comment by shawnlobo96** [comment]
@johnjbarton for tests run via the JS testing API, limited logs are captured on Browserstack's end.

All the logs are generated in the terminal or console on which the test is running via the runner.

Can you share the console logs for the failed runs with support@browserstack.com?

**Comment by johnjbarton** [comment]
Oh sure, here is typical output:
```
23 12 2020 23:05:09.417:INFO [karma-server]: Karma v5.2.3 server started at http://localhost:9876/
23 12 2020 23:05:09.417:INFO [launcher]: Launching browsers bs_chrome, bs_firefox, bs_safari, bs_ie_11, bs_ie_10, bs_ie_9 with concurrency 3
23 12 2020 23:05:09.448:INFO [launcher]: Starting browser chrome (Windows 10) on BrowserStack
23 12 2020 23:05:09.451:INFO [launcher]: Starting browser firefox (Windows 10) on BrowserStack
23 12 2020 23:05:09.451:INFO [launcher]: Starting browser safari 9.0 (OS X El Capitan) on BrowserStack
23 12 2020 23:05:19.934:INFO [launcher.browserstack]: safari 9.0 (OS X El Capitan) session at https://automate.browserstack.com/builds/020fbd32118c5950abb602fbd2fbb99ca81e44d0/sessions/7b03f1bdd0cf91a7d98a2f650ff12e7969b14441
23 12 2020 23:05:19.934:INFO [launcher.browserstack]: firefox (Windows 10) session at https://automate.browserstack.com/builds/020fbd32118c5950abb602fbd2fbb99ca81e44d0/sessions/dc7c7171d69756f0e18c0665e0152eff2a5289b2
23 12 2020 23:05:19.935:INFO [launcher.browserstack]: chrome (Windows 10) session at https://automate.browserstack.com/builds/020fbd32118c5950abb602fbd2fbb99ca81e44d0/sessions/28e0d1f78cc8f17ec22aa93e374ac0ab15ddeab5
23 12 2020 23:06:32.810:INFO [Chrome 87.0.4280.66 (Windows 10)]: Connected on socket nB8yEsApqCUBB9zQAAAB with id 34086830
...........................................
Chrome 87.0.4280.66 (Windows 10): Executed 43 of 43 SUCCESS (0.17 secs / 0.061 secs)
23 12 2020 23:06:35.472:INFO [launcher]: Starting browser ie 11.0 (Windows 10) on BrowserStack
23 12 2020 23:06:35.556:INFO [Firefox 84.0 (Windows 10)]: Connected on socket iBWEdWarrXCusc_IAAAD with id 50852622
...........................................
Firefox 84.0 (Windows 10): Executed 43 of 43 SUCCESS (0.289 secs / 0.092 secs)
23 12 2020 23:06:39.219:INFO [launcher]: Starting browser ie 10.0 (Windows 8) on BrowserStack
23 12 2020 23:06:41.958:INFO [launcher.browserstack]: ie 11.0 (Windows 10) session at https://automate.browserstack.com/builds/020fbd32118c5950abb602fbd2fbb99ca81e44d0/sessions/3d344b10776b52a11b68d92e01782c5bdba647cc
23 12 2020 23:06:41.958:INFO [launcher.browserstack]: ie 10.0 (Windows 8) session at https://automate.browserstack.com/builds/020fbd32118c5950abb602fbd2fbb99ca81e44d0/sessions/72f413b973aabc0c19fe54e86c34a7ccc4614342
23 12 2020 23:10:30.262:WARN [launcher.browserstack]: safari 9.0 (OS X El Capitan) has not captured in 300000 ms, killing.
23 12 2020 23:10:30.480:INFO [launcher]: Starting browser ie 9.0 (Windows 7) on BrowserStack
23 12 2020 23:10:37.741:INFO [launcher.browserstack]: ie 9.0 (Windows 7) session at https://automate.browserstack.com/builds/020fbd32118c5950abb602fbd2fbb99ca81e44d0/sessions/a43ca227a8ec53debf15460682c6b8d4a3b85add
23 12 2020 23:10:37.742:INFO [launcher.browserstack]: safari 9.0 (OS X El Capitan) session at https://automate.browserstack.com/builds/020fbd32118c5950abb602fbd2fbb99ca81e44d0/sessions/a28d0962fd815950e1f8529ff71c9e321fcfa3ca
23 12 2020 23:12:12.716:WARN [launcher.browserstack]: ie 11.0 (Windows 10) has not captured in 300000 ms, killing.
23 12 2020 23:12:12.718:WARN [launcher.browserstack]: ie 10.0 (Windows 8) has not captured in 300000 ms, killing.
```

**Comment by johnjbarton** [comment]
Based on the video:
https://automate.browserstack.com/builds/de1b91638cfa306c2d7f0c63c9906ac6ad18976e/sessions/cddb2412196d68f7298022b493408ee53ccfd897
The karma client is not able to connect back to the server for the failing cases.

**Comment by Namrata-m13** [comment]
@johnjbarton Can you share the above session as well as console logs for the failed runs with support@browserstack.com?

**Comment by johnjbarton** [comment]
Here is a couple of the console logs
https://travis-ci.org/github/karma-runner/karma/jobs/749870278
https://travis-ci.org/github/karma-runner/karma/jobs/753244593
Note much there.

I don't know what you mean by "share the above session". I posted the link.  What else can I do?

**Comment by bcaudan** [comment]
I have the same issue, after a bump to karma@6.0.1, IE and safari tests are failing.
Any update on this topic?

**Comment by Namrata-m13** [comment]
@bcaudan @johnjbarton karma-runner had bugs in Safari and IE from the 6.0.1 version. It has been fixed in 6.0.4. We ran some tests on v6.0.4 and everything looks fine.
https://github.com/karma-runner/karma/releases/tag/v6.0.4

**Comment by johnjbarton** [comment]
In karma 6 we broke Safari and IE, at the same time as the socket.io change but not caused by it. Sadly we released the fail because we had disabled the browserstack tests for exactly those browsers.  I spent some time to try to fix the tests first, but to be honest I could find no information on debugging browserstack tests. 

@devoto13 figured it out so our browserstack tests are working again. Thanks, and sorry for the red herring.
